### PR TITLE
drivers: pinctrl: add TI CC32XX driver 

### PR DESCRIPTION
**Pull request by gmarull**
Add a new pinctrl driver for TI CC32XX SoC. The driver has not been
tested, just implemented following datasheet specs and checked that it
compiles. Consider this as a best-effort driver to remove custom pinmux
code in board files.

This PR is part of a series to cleanup SYS_INIT usage.

### CODE REVIEW

**Comment by vaishnavachath** [comment]
Thank you for the pull request @gmarull , I will test these changes later this week and update.
Sorry for not performing this cleanup earlier, I started working on this earlier[0], but did not get the bandwidth to complete and test the changes.
0 - https://github.com/vaishnavachath/zephyr/commit/60471f04756274a2e6788fb0bd43947aae9c3314

**Review by henrikbrixandersen — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]
It's such a wild concept to see Nordic copyright on code for a TI platform, haha..

LGTM, but I would prefer @vaishnavachath to have the final say.

And, obviously, hold off merging until we've tagged the release.

**Review by vaishnavachath — APPROVED** [review]

**Pull request merged**
